language/stdlib/modules/libra_configs.move

```diff
@@ -31,38 +37,58 @@ module LibraConfig {
         });
     }
 
-    // We can't return reference here, this returns a copy of the config value.
-    public fun get<Config: copyable>(addr: address): Config acquires T {
+    public fun apply_for_default_privilege() {
+        Association::apply_for_association();
+        Association::apply_for_privilege<ConfigCreatorPrivilege>();
+    }
+
+    public fun grant_default_privilege(addr: address) {
+        Association::grant_association_address(addr);
+        Association::grant_privilege<ConfigCreatorPrivilege>(addr);
+    }
+
+    // Get a copy of `Config` value stored under `addr`.
+    public fun get<Config: copyable>(): Config acquires T {
+        let addr = default_config_address();
         Transaction::assert(::exists<T<Config>>(addr), 24);
         *&borrow_global<T<Config>>(addr).payload
     }
 
     // Set a config item to a new value and trigger a reconfiguration.
-    public fun set<Config: copyable>(addr: address, payload: Config) acquires T, Configuration {
-        // TODO: impose proper permission checks
-        // Callable by the any address for now.
-
+    public fun set<Config: copyable>(_cap: &ConfigModifierPrivilege<Config>, payload: Config) acquires T, Configuration {
+        let addr = default_config_address();
         Transaction::assert(::exists<T<Config>>(addr), 24);
         let config = borrow_global_mut<T<Config>>(addr);
         config.payload = payload;
 
         reconfigure_();
     }
 
-    // Publish a new config item to a new value and trigger a reconfiguration.
-    public fun publish_new_config<Config: copyable>(payload: Config) {
-        // TODO: impose proper permission checks
-        // Callable by the any address for now.
+    // Publish a new config item.
+    public fun publish_new_config<Config: copyable>(payload: Config): ConfigModifierPrivilege<Config> {
+        Transaction::assert(
```

Comment:
Do you mean `publish<T>(payload);  set<T>(cap, T)`? `get` should always succeed because get is always reading the config data from the config address.

Reply:
> Do you mean publish<T>(payload); set<T>(cap, T)

Yes, my bad.

> get should always succeed because get is always reading the config data from the config address.

But what if an address other than 0xF1A95 calls `publish_new_config`?

Reply:
Do you think we should allow that? In my previous mindset I thought accounts with such capability should be able to publish new config. This allows different configs to resides in different addresses, which can be good or bad.

Reply:
I don't think we should allow it, but without the explicit address check it is allowed :)